### PR TITLE
refactor: widgets

### PR DESCRIPTION
**Pull request by 634750802**
- widget: use jsx instead of factory for compose widget visualization
- cli: generate files in cli, remove `widgets-next` package
- endpoints: generate new dynamic ~~`@ossinsight/endpoints`~~ `@ossinsight/data-service/endpoints` package for code splitting
- datasource: support `endpoint` type
- refine `@ossinsight/widgets`

#### `@ossinsight/data-service/endpoints`

image
image

#### `@ossinsight/widgets`

image

### CODE REVIEW

File: packages/endpoints-core/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@ossinsight/endpoints-core",
```
Comment (Mini256) [inline packages/endpoints-core/package.json]: Maybe we can call it '@ossinsight/data-service'? 😄

File: web/app/api/queries/[...query]/route.ts [hunk 2]
```diff
@@ -1,33 +1,46 @@
...
+export const runtime = 'edge';
 
-const dataService = new DataService({
-  url: process.env.DATABASE_URL
-});
-
-export async function GET(req: NextRequest) {
+export async function GET (req: NextRequest) {
   // Remove prefix.
   const queryName = req.nextUrl.pathname.replaceAll('/api/queries/', '');
```
Comment (Mini256) [inline web/app/api/queries/[...query]/route.ts]:
```suggestion
  const queryName = params.query.map(decodeURIComponent).join('/');
```

File: web/app/api/queries/[...query]/route.ts [hunk 3]
```diff
@@ -1,33 +1,46 @@
...
 
-export const runtime = 'edge'
+export const runtime = 'edge';
 
-const dataService = new DataService({
-  url: process.env.DATABASE_URL
-});
-
-export async function GET(req: NextRequest) {
+export async function GET (req: NextRequest) {
```
Comment (Mini256) [inline web/app/api/queries/[...query]/route.ts]:
```suggestion
export interface IParams {
  query: string[];
}

export async function GET (req: NextRequest, { params }: { params: IParams }) {
```

File: packages/cli/templates/endpoint.js.liquid [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export const sql = {{ sql | json }};
```
Comment (Mini256) [inline packages/cli/templates/endpoint.js.liquid:1]:
```suggestion
export const sql = {{ sql | json }};
```

File: packages/endpoints-core/package.json (hunk 1 above)
Comment (634750802) [reply packages/endpoints-core/package.json]: changed

**Review by shczhen — APPROVED** [review]

**Pull request merged**